pitmutationmate/src/main/kotlin/com/amos/pitmutationmate/pitmutationmate/execution/BasePitestExecutor.kt

```diff
@@ -45,10 +46,15 @@ abstract class BasePitestExecutor {
                 // update tool window with latest result data
                 val toolWindow: ToolWindow? = ToolWindowManager.getInstance(project).getToolWindow(ToolWindowFactory.ID)
                 // safe and get latest pitest results and update report toolWindow with it
-                val coverageReport = project.service<MutationResultService>().updateLastMutationResult()?.coverageReports?.first()
+                val resultData = project.service<MutationResultService>().updateLastMutationResult()
+                val coverageReport = resultData?.coverageReports?.first()
                 if (toolWindow != null) {
                     ToolWindowFactory.Util.updateReport(toolWindow, coverageReport)
                 }
+                // archive the pitestrun using the runarchiver
+                if (classFQN != null) {
+                    RunArchiver(classFQN, project).archiveRun()
```

Comment:
Might be a good idea to run this asynchronous because for large files it could take a sec to move them